tests/cnf/core/network/metallb/tests/frrk8-tests.go

```diff
@@ -351,10 +353,15 @@ var _ = Describe("FRR", Ordered, Label(tsparams.LabelFRRTestCases), ContinueOnFa
 				By("Verify node state updates on worker node 0")
 				Eventually(func() string {
 					// Get the routes
-					frrNodeState, err := metallb.ListFrrNodeState(APIClient)
+					frrNodeState, err := metallb.ListFrrNodeState(APIClient, client.ListOptions{
+						FieldSelector: fields.SelectorFromSet(fields.Set{"metadata.name": "worker-0"})})
 					Expect(err).ToNot(HaveOccurred(), "Failed to verify BGP routes")
 
-					return frrNodeState[0].Object.Status.RunningConfig
+					frrWorker0Status := findNodeByName(frrNodeState, cnfWorkerNodeList[0].Definition.Name)
+					Expect(frrWorker0Status).ToNot(BeNil(), fmt.Sprintf("Node with name %s not found",
+						cnfWorkerNodeList[0].Definition.Name))
+
+					return frrWorker0Status.Object.Status.RunningConfig
```

Comment:
`return frrNodeState[0].Object.Status.RunningConfig`

Back to original

Reply:
changed